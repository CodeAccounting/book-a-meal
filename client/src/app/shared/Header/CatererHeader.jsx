import React, { Fragment } from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

/**
 * @exports
 * @function CatererHeader
 * @param {object} props
 * @returns {JSX} CatererHeader
 */
const CatererHeader = ({ date, showTime }) => (
  <Fragment>
    <div className="menu-control d-none-md">
      <a href="#sidenav" className="d-none-md">
        <img src="./img/toggler.png" alt="toggler" />
      </a>
    </div>
    <div className="page-title d-flex-md">
      <div className="date">
        <h2>{date}&nbsp; &nbsp; {showTime && moment().format('HH:mm')}</h2>
      </div>
    </div>
  </Fragment>
);

CatererHeader.propTypes = {
  date: PropTypes.string,
  showTime: PropTypes.bool
};

CatererHeader.defaultProps = {
  date: moment().format('dddd[,] Do MMMM YYYY'),
  showTime: true
};

export default CatererHeader;
