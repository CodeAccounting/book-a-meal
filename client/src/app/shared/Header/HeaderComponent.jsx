import React from 'react';

const Header = () => (
  <div className="header homepage">
    <div className="navbar unauth-navbar">
      <div className="page-title">
        <h3>
          <a href="/">BOOK-A-MEAL</a>
        </h3>
      </div>
      <div className="navlinks">
        <a href="./login.html" className="link">Log In</a>
        <a href="./signup.html" className="link">Sign Up</a>
      </div>
    </div>
  </div>
);

export default Header;
