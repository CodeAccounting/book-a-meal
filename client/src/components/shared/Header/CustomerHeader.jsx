import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import LinkBtn from '../Link';
import { userPropTypes } from '../../../helpers/proptypes';
import Dropdown from '../Dropdown';


/**
 * @exports
 * @function CustomerHeader
 * @param {object} props
 * @returns {JSX} CustomerHeader
 */
const CustomerHeader = ({ logout, user, active }) => (
  <Fragment>
    <div className="page-title">
      <h3>
        <a href="/">BOOK-A-MEAL</a>
      </h3>
    </div>
    <div className="navlinks">
      <LinkBtn href="#" className="link">
        <img src="/img/cart.png" alt="cart" />
      </LinkBtn>
      <div className="user-info">
        <Dropdown
          type="customer"
          toggler={
            <div className="username-circle circle-click">
              <p className="circle-click">{user.firstname[0]}&#9663;</p>
            </div>
          }
          content={
            <Fragment>
              <a href="/" className={`${active === 'menu' && 'active'}`}>Menu</a>
              <a href="/">Order History</a>
              <a href="/" onClick={logout}>Logout</a>
            </Fragment>
          }
        />
        <p className="hide d-lg">{user.firstname}</p>
      </div>
    </div>
  </Fragment>
);

CustomerHeader.propTypes = {
  ...userPropTypes,
  logout: PropTypes.func.isRequired,
  active: PropTypes.string.isRequired
};

export default CustomerHeader;
