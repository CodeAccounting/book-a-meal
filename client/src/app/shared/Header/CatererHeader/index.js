import { connect, } from 'react-redux';
import CatererHeader from './CatererHeader';

const mapStateToProps = state => ({
  currentDay: state.menu.currentDay
});

export default connect(mapStateToProps)(CatererHeader);
