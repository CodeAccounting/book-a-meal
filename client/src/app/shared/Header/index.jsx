import React, { Component, Fragment } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import CatererHeader from './CatererHeader';

/**
 * @exports
 * @class Header
 * @extends Component
 * @classdesc Creates Header Component
 * @returns {JSX} Header Component
 */
class Header extends Component {
  /**
   * Renders the Home Header
   * @memberof Dashboard
   * @returns {JSX} Dashboard Component
   */
  renderUnauthHeader = () => {
    const { type } = this.props;

    return (
      <Fragment>
        <div className="page-title">
          <h3>
            <a href="/">BOOK-A-MEAL</a>
          </h3>
        </div>
        {type === 'home' &&
        <div className="navlinks">
          <a href="./signin" className="link">Log In</a>
          <a href="./signup" className="link">Sign Up</a>
        </div>}
      </Fragment>
    );
  }
  /**
   * @memberof Header
   * @returns {JSX} Header Component
   */
  render() {
    const { type } = this.props;
    const unauth = type === 'home' || type === 'unauth';

    const headerClass = classNames({
      header: true,
      homepage: type === 'home',
      unauth: type === 'unauth'
    });

    const navbarClass = classNames({
      navbar: true,
      'unauth-navbar': unauth,
    });

    return (
      <div className={headerClass}>
        <div className={navbarClass}>
          {unauth &&
          this.renderUnauthHeader()}
          {type === 'caterer' &&
          <CatererHeader />}
        </div>
      </div>
    );
  }
}

Header.propTypes = {
  type: PropTypes.string
};

Header.defaultProps = {
  type: null
};

export default Header;
