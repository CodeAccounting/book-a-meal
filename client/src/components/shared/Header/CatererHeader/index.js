import { connect, } from 'react-redux';
import { bindActionCreators } from 'redux';
import CatererHeader from './CatererHeader';
import { toggleSideNav } from '../../../../store/actions/ui';

const mapStateToProps = state => ({
  currentDay: state.menu.currentDay
});

const mapDispatchToProps = dispatch => bindActionCreators({
  toggleSideNav
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(CatererHeader);
