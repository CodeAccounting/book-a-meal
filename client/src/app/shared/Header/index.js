import { connect } from 'react-redux';
import HeaderComponent from './HeaderComponent';

const mapStateToProps = state => ({
  type: state.router.location.pathname,
});

export default connect(mapStateToProps)(HeaderComponent);
