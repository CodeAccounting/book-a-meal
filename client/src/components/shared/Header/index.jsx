import React, { Component, Fragment } from 'react';
import { Link } from 'react-router-dom';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import CatererHeader from './CatererHeader';
import CustomerHeader from './CustomerHeader';
import { userPropType } from '../../../helpers/proptypes';
import './Header.scss';

/**
 * @exports
 * @class Header
 * @extends Component
 * @classdesc Creates Header Component
 * @returns {JSX} Header Component
 */
class Header extends Component {
  /**
   * Renders the Home Header
   * @memberofHeader
   * @returns {JSX}Header Component
   */
  renderUnauthHeader = () => {
    const { type } = this.props;

    return (
      <Fragment>
        <div className="page-title">
          <h3>
            <a href="/">BOOK-A-MEAL</a>
          </h3>
        </div>
        {type === 'home' &&
          <div className="navlinks">
            <Link href="/signin" to="/signin" className="link">Log In</Link>
            <Link href="/signup" to="/signup" className="link">Sign Up</Link>
          </div>}
      </Fragment>
    );
  }
  /**
   * @memberof Header
   * @returns {JSX} Header Component
   */
  render() {
    const { type, showTime, dateType } = this.props;
    const unauth = type === 'home' || type === 'unauth';

    const headerClass = classNames({
      header: true,
      homepage: type === 'home',
      unauth: type === 'unauth'
    });

    const navbarClass = classNames({
      navbar: true,
      'unauth-navbar': unauth,
      'white-navbar': type === 'customer'
    });

    return (
      <div className={headerClass}>
        <div className={navbarClass}>
          {unauth &&
            this.renderUnauthHeader()}
          {type === 'caterer' &&
            <CatererHeader dateType={dateType} showTime={showTime} />}
          {type === 'customer' &&
            <CustomerHeader {...this.props} />}
        </div>
      </div>
    );
  }
}

Header.propTypes = {
  user: userPropType,
  type: PropTypes.string,
  showTime: PropTypes.bool,
  dateType: PropTypes.string,
  logout: PropTypes.func,
};

Header.defaultProps = {
  user: null,
  type: null,
  showTime: true,
  dateType: null,
  logout: null
};

export default Header;
