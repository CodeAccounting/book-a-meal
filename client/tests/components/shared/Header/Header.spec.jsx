import React from 'react';
import Header from '../../../../src/app/shared/Header';

describe('Header', () => {
  it('renders correctly', () => {
    const shallowWrapper = shallow(<Header type="home" />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('renders unauthenticated header correctly', () => {
    const shallowWrapper = shallow(<Header type="unauth" />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('renders caterer header correctly', () => {
    const shallowWrapper = shallow(<Header type="caterer" />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});
