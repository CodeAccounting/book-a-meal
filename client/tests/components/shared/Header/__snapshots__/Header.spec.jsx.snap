// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders caterer header correctly 1`] = `
<div
  className="header"
>
  <div
    className="navbar"
  >
    <Connect(CatererHeader)
      dateType={null}
      showTime={true}
    />
  </div>
</div>
`;

exports[`Header renders correctly 1`] = `
<div
  className="header homepage"
>
  <div
    className="navbar unauth-navbar"
  >
    <React.Fragment>
      <div
        className="page-title"
      >
        <h3>
          <a
            href="/"
          >
            BOOK-A-MEAL
          </a>
        </h3>
      </div>
      <div
        className="navlinks"
      >
        <Link
          className="link"
          href="/signin"
          replace={false}
          to="/signin"
        >
          Log In
        </Link>
        <Link
          className="link"
          href="/signup"
          replace={false}
          to="/signup"
        >
          Sign Up
        </Link>
      </div>
    </React.Fragment>
  </div>
</div>
`;

exports[`Header renders unauthenticated header correctly 1`] = `
<div
  className="header unauth"
>
  <div
    className="navbar unauth-navbar"
  >
    <React.Fragment>
      <div
        className="page-title"
      >
        <h3>
          <a
            href="/"
          >
            BOOK-A-MEAL
          </a>
        </h3>
      </div>
    </React.Fragment>
  </div>
</div>
`;
